### PR TITLE
Add Code Coverage For ToolStrip

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

## Proposed changes
- Add unit tests for ToolStrip to test its properties: Visible, Size, BackgroundImage, CausesValidation, ContextMenuStrip, CausesValidation, Cursor, ForeColor
- Add unit tests for ToolStrip to test its methods: Compare(), Show()
- Add unit tests for ToolStrip to test its events: DockChanged, BackColorChanged, BackgroundImageChanged, BackgroundImageLayoutChanged, BottomToolStripPanelVisible,CausesValidationChanged, ContextMenuStripChanged, CursorChanged
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11464)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs [hunk 1]
```diff
@@ -92,4 +93,33 @@ public void ToolStripDropDownButtonAccessibleObject_FragmentNavigate_Child_Retur
         Assert.Equal(expected, accessibleObject.FragmentNavigate(NavigateDirection.NavigateDirection_FirstChild));
         Assert.Equal(expected, accessibleObject.FragmentNavigate(NavigateDirection.NavigateDirection_LastChild));
     }
+
+    private ToolStripDropDownButton CreateToolStripDropDownButton(string text = null, Bitmap image = null, EventHandler onClick = null, string name = null, ToolStripItem[] dropDownItems = null)
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs:97]: This seems to only be used in the test below. Consider making this a local function

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs [hunk 2]
```diff
@@ -92,4 +93,33 @@ public void ToolStripDropDownButtonAccessibleObject_FragmentNavigate_Child_Retur
...
+        toolStripDropDownButton.Name = name;
+
+        return toolStripDropDownButton;
+    }
+
+    [WinFormsFact]
+    public void ToolStripDropDownButton_ConstructorParameters_ShouldInitializeCorrectly()
+    {
+        string text = "Test text";
+        Bitmap image = new(10, 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs:110]:
```suggestion
        using Bitmap image = new(10, 10);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs [hunk 3]
```diff
@@ -92,4 +93,33 @@ public void ToolStripDropDownButtonAccessibleObject_FragmentNavigate_Child_Retur
...
+        bool wasClicked = false;
+        EventHandler onClick = (sender, e) => { wasClicked = true; };
+        string name = "Test name";
+        var dropDownItems = new ToolStripItem[] { new ToolStripMenuItem("Test item") };
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObjectTests.cs:114]: I believe we can initialize this one like so
```suggestion
        ToolStripItem[] dropDownItems = [new ToolStripMenuItem("Test item")];
```

File: src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs [hunk 4]
```diff
@@ -1,27 +1,216 @@
...
+        _toolStripContainer.BackColorChanged += handler;
+        _toolStripContainer.BackColorChanged -= handler;
+
+        callCount.Should().Be(0);
+    }
+
+    [WinFormsFact]
+    public void ToolStripContainer_BackgroundImage_GetSet_ReturnsExpected()
+    {
+        Image value = new Bitmap(10, 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs:67]: Will the `ToolStripContainer` dispose this? If not, please add a using

File: src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs [hunk 5]
```diff
@@ -1,27 +1,216 @@
...
+
+    [WinFormsFact]
+    public void ToolStripContainer_BackColorChanged_AddRemoveEvent_MaintainsExpected()
+    {
+        int callCount = 0;
+        EventHandler handler = (sender, e) => callCount++;
+        _toolStripContainer.BackColorChanged += handler;
+        _toolStripContainer.BackColorChanged -= handler;
+
+        callCount.Should().Be(0);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs:61]: Maybe also can test when the value of callCount will be 1.

File: src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs [hunk 6]
```diff
@@ -1,27 +1,216 @@
...
+
+    [WinFormsFact]
+    public void ToolStripContainer_BackgroundImageChanged_AddRemoveEvent_MaintainsExpected()
+    {
+        int callCount = 0;
+        EventHandler handler = (sender, e) => callCount++;
+        _toolStripContainer.BackgroundImageChanged += handler;
+        _toolStripContainer.BackgroundImageChanged -= handler;
+
+        callCount.Should().Be(0);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs:81]: As above.

File: src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs [hunk 7]
```diff
@@ -1,27 +1,216 @@
...
+
+    [WinFormsFact]
+    public void ToolStripContainer_BackgroundImageLayoutChanged_AddRemoveEvent_MaintainsExpected()
+    {
+        int callCount = 0;
+        EventHandler handler = (sender, e) => callCount++;
+        _toolStripContainer.BackgroundImageLayoutChanged += handler;
+        _toolStripContainer.BackgroundImageLayoutChanged -= handler;
+
+        callCount.Should().Be(0);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/ToolStripContainerTests.cs:92]: As above.

**Review by SimonZhao888 — APPROVED** [review]

**Pull request merged**